scipy/stats/_entropy.py

```diff
@@ -196,7 +257,60 @@ def differential_entropy(
     logs = np.log(
         len(differences) * differences / (2 * window_length),
     )
-    if base is not None:
-        logs /= np.log(base)
 
     return np.mean(logs, axis=0)
+
+
+def _pad_along_last_axis(X, m):
+    """Pad the data for computing the rolling window difference"""
+    # scales a  bit better than method in _vasicek_like_entropy
+    shape = np.array(X.shape)
+    shape[-1] = m
+    Xl = np.broadcast_to(X[..., [0]], shape)  # [0] vs 0 to maintain shape
+    Xr = np.broadcast_to(X[..., [-1]], shape)
+    return np.concatenate((Xl, X, Xr), axis=-1)
+
+
+def _van_es_entropy(X, m):
+    """Compute the van Es estimator as described in [6]"""
+    # No equation number, but referred to as HVE_mn.
+    # Typo: there should be a log within the summation.
+    n = X.shape[-1]
+    term1 = 1/(n-m) * np.sum(np.log((n+1)/m * (X[..., m:] - X[..., :-m])),
+                             axis=-1)
+    k = np.arange(m, n+1)
+    return term1 + np.sum(1/k) + np.log(m) - np.log(n+1)
```

Comment:
I'm not sure I understand this one. What extra spaces? Is it whitespace inside the function or double blank line after the function? The latter is a PEP8 requirement.

Reply:
Oops, I understand now after reading [this](https://github.com/scipy/scipy/pull/13845/files#r611473529). Which would you prefer? I suppose i could do:
```suggestion
    difference = X[..., m:] - X[..., :-m]
    term1 = 1/(n-m) * np.sum(np.log((n+1)/m * difference), axis=-1)
    k = np.arange(m, n+1)
    return term1 + np.sum(1/k) + np.log(m) - np.log(n+1)
```

Reply:
I wanted to add extra space between the operations so `/ - +`. Here I am not sure we need the variable `difference`. Below I was actually thinking that it might be slower to have this extra variable.

Reply:
Referring to something that has to be calculated takes ~0s extra time.
```
> x = np.random.rand(100, 100, 100, 100)
> %timeit difference = x - x
205 ms ± 1.21 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
> %timeit x - x
208 ms ± 5.75 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```